### ISSUE TITLE
[5.7.0] ExecutionFailed: bitIndex negative

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13169]
When running the following query:
```cypher
CALL { MERGE () } RETURN null AS n0 UNION ALL MATCH () MATCH ()<-[:((!A&B)&(C|D))]-() RETURN null AS n0
```

I encountered the following error:
```
Neo4jError: Neo.DatabaseError.Statement.ExecutionFailed (bitIndex < 0: -1)
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.

I encountered this issue when testing queries against the **Neo4j 5.7.0 enterprise version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following query and observe it throws an error:
```cypher
CALL { MERGE () } RETURN null AS n0 UNION ALL MATCH () MATCH ()<-[:((!A&B)&(C|D))]-() RETURN null AS n0
```

### Expected behavior
The query should run successfully

### Actual behavior
Neo4j 5.6.0: The query runs successfully
Neo4j 5.7.0:
- **Community:** The query runs successfully
- **Enterprise:** The query fails with the error message `Neo4jError: Neo.DatabaseError.Statement.ExecutionFailed (bitIndex < 0: -1)`

**Comment by loveleif** [comment]
Thanks @DominicWuest, you were fast to adopt 5.7.0!

I can reproduce this issue, it looks to be a problem with the task scheduling of pipelined runtime. A workaround is to prefix query with `CYPHER runtime=slotted`. We're prioritising this issue and will get back when it's fixed ([link to internal bug tracker](https://trello.com/c/0t5rPOE5)).

I suspect the issue is present also in 5.6.0 for other queries, but this particular query could have gotten planned differently.

Have a nice weekend.

**Comment by loveleif** [comment]
Thanks again! This issue have been fixed and is scheduled to be released in 5.8.0. I have not been able to reproduce the issue in 4.4. Please create a new issue if the problem remains in 5.8.0/4.4.* despite our best efforts.
